src/uct/tcp/tcp_ep.c

```diff
@@ -1074,6 +1074,8 @@ uct_tcp_ep_am_prepare(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
         if (ucs_likely(status == UCS_ERR_NO_RESOURCE)) {
             goto err_no_res;
         }
+
+        uct_tcp_ep_set_failed(ep);
```

Comment:
can remove, should return CONNECTION_RESET if state is closed
err callback needs to be called only once

Reply:
done